designs/native-ami-versioning.md

```diff
@@ -0,0 +1,121 @@
+# Native AMI Versioning
+
+Karpenter uses the latest EKS Optimized AMI releases as defaults in their AMI Requirements. With [Drift](https://karpenter.sh/preview/concepts/deprovisioning/#drift) implemented, when a new EKS Optimized AMI is released users who have enabled drift could see their fleet rolled out to the newly released AMI without warning.
+
+[EKS Optimized AMI Releases](#eks-optimized-ami-releases) are irregular and versioned differently. Without validation, irregular release cadences could introduce unwanted AMIs in production. This doc proposes two functional improvements to AMI selection logic to ease customer pain.
+
+## What could happen when a new EKS Optimized AMI is released?
+
+When a new EKS Optimized AMI is released, Karpenter could begin deprovisioning and replacing nodes. If a user wants to revert back to an older release, they need to find the corresponding AMI IDs and change their AWSNodeTemplate to pin those values by setting their AWSNodeTemplate to the following:
+
+```
+spec:
+    amiSelector:
+        aws-ids: "ami-123,ami-456"
+```
+
+Once the user wants to return to the latest release, they’ll have to restore their configurations to the previous default as the following.
+
+```
+spec:
+    amiFamily: Ubuntu # Or Bottlerocket or AL2
+```
+
+While this isn’t a difficult process, it isn't viable to require users to jump through these hoops. They must be aware of new EKS Optimized AMI releases, understand SSM aliases to fetch the AMI IDs, and modify their AWSNodeTemplate values in their CI/CD pipelines multiple times. Some users have [requested these improvements](https://github.com/aws/karpenter/issues/1495). In order to improve the experience for all users, we’ll introduce (1) Bake Time and (2) Release Version Pinning improvements.
+
+## Bake Time
+
+### 🔑 Introduce a minimumAgeDays field in the amiSelector.
+
+While this hasn’t been asked for explicitly, bake time is a common concept in phasing in new changes. Whether a user has a mandated bake time for production or developers have an estimated time to validate workloads in before rolling out to prod, bake time provides a sliding window of leeway. Users can be cautious and set a high value, or leave out the setting for no bake time requirements.
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  amiSelector:
+    minimumAgeDuration: 10
```

Comment:
I thought we had talked about duration! :) Was the above just outdated?

Reply:
Yes! Sorry about that.